hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/webapp/util/WebAppUtils.java

```diff
@@ -428,6 +429,12 @@ public static String getAggregatedLogURL(String serverHttpAddress,
         user == null || user.isEmpty()) {
       return null;
     }
+    String [] ipAdds = serverHttpAddress.split("//");
+    if (StringUtils.countMatches(ipAdds[1], ":") > 2) {
```

Comment:
Should we check the length of this array after split, and NPE may occur if we use it directly?

Reply:
Thanks for review, 
modified as you mentioned.